spec/models/reports/coda_finance_report_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+RSpec.describe Reports::CodaFinanceReport do
+  let(:home_office) { FactoryBot.create(:customer, :central_government, name: 'Home Office') }
+  let(:framework) { FactoryBot.create(:framework, coda_reference: 409999, name: 'G CLOUD', short_name: 'RM3787') }
+  let(:task_1) do
+    FactoryBot.create(:task, framework: framework, supplier: supplier_1, period_month: 8, period_year: 2018)
+  end
+  let(:task_2) do
+    FactoryBot.create(:task, framework: framework, supplier: supplier_2, period_month: 8, period_year: 2018)
+  end
+  let(:supplier_1) { FactoryBot.create(:supplier, name: 'Bob', coda_reference: 'C099999') }
+  let(:supplier_2) { FactoryBot.create(:supplier, name: 'Mary', coda_reference: 'C011111') }
+
+  let(:no_business_submission) do
+    FactoryBot.create(:no_business_submission, framework: framework, supplier: supplier_1, task: task_1)
+  end
+  let(:submission) do
+    FactoryBot.create(
+      :submission,
+      entries: [submission_entry_1, submission_entry_2],
+      supplier: supplier_2,
+      framework: framework,
+      task: task_1
+    )
+  end
+  let(:submission_entry_1) do
+    FactoryBot.build(
+      :validated_submission_entry,
+      data: { 'Total Cost (ex VAT)' => '678.55', 'Customer URN' => home_office.urn },
+      source: { sheet: 'InvoicesRaised', row: 1 }
+    )
+  end
+  let(:submission_entry_2) do
+    FactoryBot.build(
+      :validated_submission_entry,
+      data: { 'Total Cost (ex VAT)' => '123.45', 'Customer URN' => home_office.urn },
+      source: { sheet: 'InvoicesRaised', row: 2 }
+    )
+  end
+
+  let(:submissions) { [submission, no_business_submission] }
+  let(:report) { Reports::CodaFinanceReport.new(submissions) }
+  let(:report_timestamp) { Time.zone.now.to_i }
+
+  let(:expected_csv) do
+    <<~CSV
+      RunID,Nominal,Customer Code,Customer Name,Contract ID,Order Number,Lot Description,Inf Sales,Commission,Commission %,End User,Submitter,Month,M_Q
+      DSS-#{report_timestamp},409999,C011111,Mary,RM3787,,G CLOUD,802.00,12.03,0.015,UCGV,Mary,August 2018,M
+      DSS-#{report_timestamp},409999,C011111,Mary,RM3787,,G CLOUD,0.00,0.00,0.015,UWPS,Mary,August 2018,M
```

Comment:
Where is this entry coming from? I can only see the `home_office` customer (with a URN assigned as Central Government) in the spec.

Also, those no business submissions surely wouldn't have a sector, since there's no entries to determine it with?

Reply:
Hmmm, so this is based on a possibly incorrect assumption that we should report a row regardless of whether their is management charge or not. I based my assumption on the bit about the files that should be produced, including: "A file containing submissions with spend data which won’t lead to an invoice". I interpreted that as meaning no business returns, but reading it now, that is probably wrong...